src/Trains.NET.Engine/GamePlay/GameManager.cs

```diff
@@ -71,6 +74,8 @@ public void GameLoopStep()
             foreach (var gameStep in _gameSteps)
             {
                 gameStep.Update(timeSinceLastTick);
+                if(_gameStateManager.SaveMode == SaveModes.GameStep)
+                    _gameStateManager.Save();
```

Comment:
So the galaxy brain move here would to have game state manager implement IGameStep, and then it would automatically be injected and called without any other changes 😁